aptos-move/framework/aptos-token-objects/sources/collection.move

```diff
@@ -125,23 +152,40 @@ module aptos_token_objects::collection {
         let constructor_ref = object::create_named_object(creator, collection_seed);
         let object_signer = object::generate_signer(&constructor_ref);
 
-        let supply = FixedSupply {
-            current_supply: 0,
-            max_supply,
-            total_minted: 0,
-            burn_events: object::new_event_handle(&object_signer),
-            mint_events: object::new_event_handle(&object_signer),
-        };
-
-        create_collection_internal(
-            creator,
-            constructor_ref,
-            description,
-            name,
-            royalty,
-            uri,
-            option::some(supply),
-        )
+        if (features::concurrent_token_v2_enabled()) {
+            let supply = ConcurrentSupply {
+                current_supply: aggregator_v2::create_aggregator(max_supply),
```

Comment:
Not the case currently but what would happen if we allow creators to change the max_supply. Is it possible to change it in the aggregator struct? Any implications on conflicts/concurrency etc.?

Reply:
We don't have the API for it, but we can add a native function to support it. 

fixed limit is critical to removing conflicts and allowing concurrency, so that transaction would basically be a barrier. As long as the limit is being changed one-off, that should be fine. Changing limit often - i.e. many times within the same block - would be very inefficient.

Reply:
Just a hypothetical question for token v2 as we don't have a function to update the max supply either :) Good to know it's possible if that ever happens in the future